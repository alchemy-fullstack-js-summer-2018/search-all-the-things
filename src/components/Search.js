import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class Search extends Component {

  state = {
    search: '',
    sectors: [
      'Technology',
      'Industrials',
      'Health Care',
      'Energy',
      'Financials'
    ]
  };

  static propTypes = {
    onSearch: PropTypes.func.isRequired
  };

  handleSectorSelect = (search) => {
    this.setState({ search });
  };

  handleSubmit = event => {
    event.preventDefault();
    this.props.onSearch(this.state.search);
  };

  render() {
    const { sectors } = this.state;

    return (
      <form className="search-form" onSubmit={event => this.handleSubmit(event)}>
        <select onChange={({ target }) => this.handleSectorSelect(target.value)}>
          <option value="">Select a sector</option>
          {sectors.map(sector => (
            <option key={sector} value={sector}>{sector}</option>
          ))}
        </select>
        <button>Search</button>
      </form>
    );
  }
}
