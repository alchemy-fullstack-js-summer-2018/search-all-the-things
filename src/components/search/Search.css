.search-form {
  border: 1px solid darkslategray;
  color: darkslategray;
}