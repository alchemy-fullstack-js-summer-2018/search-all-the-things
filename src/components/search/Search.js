import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Search.css';

export default class Search extends Component {

  state = {
    search: ''
  };

  static propTypes = {
    onSearch: PropTypes.func.isRequired
  };

  handleSubmit = event => {
    event.preventDefault();
    this.props.onSearch(this.state);
  };

  handleChangeSearch = ({ target }) => {
    this.setState({ search: target.value });
  };

  render() {
    const { search } = this.state;

    return ( 
      <form className="search-form" onSubmit={event => this.handleSubmit(event)}>
        <label>
        Search For:&nbsp;
          <input name="search" value={search} onChange={this.handleChangeSearch}/>
        </label>
        <button>Search</button>
      </form>
    );
  }
}
