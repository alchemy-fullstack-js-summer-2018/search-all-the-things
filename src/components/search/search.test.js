import React from 'react';
import { shallow } from 'enzyme';
import Search from './Search';
import toJSON from 'enzyme-to-json';

describe('Search', () => {

  it.skip('renders as intended', () => {
    const wrapper = shallow(<Search onSearch={() => {}}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});