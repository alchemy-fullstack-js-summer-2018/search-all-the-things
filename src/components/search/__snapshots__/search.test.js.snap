// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders as intended 1`] = `
<form
  className="search-form"
  onSubmit={[Function]}
>
  <label>
    Search For:Â 
    <input
      name="search"
      onChange={[Function]}
      value=""
    />
  </label>
  <button>
    Search
  </button>
</form>
`;
